### PR TITLE
*.csv file with cars and status

### PR DESCRIPTION
**Pull request by ghost**
created a csv-file and in the resources directory...

#21

### CODE REVIEW

File: src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import com.univocity.parsers.common.record.Record
+import com.univocity.parsers.csv.CsvParser
+import com.univocity.parsers.csv.CsvParserSettings
+import com.univocity.parsers.csv.CsvWriter
+import com.univocity.parsers.csv.CsvWriterSettings
+
+fun main(args: Array<String>) {
+
+
+    private fun readCarFromCsv(fileName: String): Car {
```
Comment (nipafx) [inline src/main/kotlin/Main.kt]: This function should not be declared _within_ the `main` function. Please move it.

Also, it says that it returns a single `Car`. But wouldn't it make more sense to return an entire list of cars?

File: src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        val settings = CsvParserSettings()
+        settings.format.setLineSeparator("\n")
+        settings.isHeaderExtractionEnabled = true
+
+        val csvParser = CsvParser(settings)
+        val reader = FileAccess().getReader(fileName)
```
Comment (nipafx) [inline src/main/kotlin/Main.kt]: The file with the class `FileAccess` is missing from this pull request.

File: src/main/kotlin/Main.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        val carRows: MutableList<Record> = csvParser.parseAllRecords(reader)
+
+        for (record in carRows) {
+            val carID = record.values.get(0)
+            val carStatus = record.values.get(1)
+
```
Comment (nipafx) [inline src/main/kotlin/Main.kt]: Now you need to actually create a `Car`:

* you can't use the `carID`, yet - that is fine
* you need a road network for that, so just create one (that part of the code will likely change in the future)

**Review by nipafx — CHANGES_REQUESTED** [review]
Good first step, here are a few things to do. Leave a comment when you're done, so we can tackle more required changes.

**Comment by ghost** [comment]
Would it be better if I close this PR, and push the *.csv file from an different branch, that this could be merged (if it is that what we need). And then we could handle them independent from each other.
What are you thinking?

**Comment by nipafx** [comment]
No reason to do that, no. The CSV file without the code reading is is pretty much useless, so we can continue in this branch. Just make the changes I requested and push the new commits to your `csv` branch.

**Comment by ghost** [comment]
Ok, I´ll try it now.

**Comment by ghost** [comment]
Oh shit! ...tried to "add" and then "commit" the changes on Git, added the class "Network", but wanted to remove that from the commiting stage cause I didn´t changed anything. Just tried something, but deleted it later. Used "git rm -f ..." . Thought it would only remove it from the stage, not from my complete branch. Is there a way to get it back?

...I remember you saying we should be careful with the command "-f"... learning by doing...
sh**

**Comment by EdenTewelde** [comment]
I don't know how to get it back, but for the next time you want to unstage something, use "git reset HEAD <file-name>"
Good luck

**Comment by nipafx** [comment]
So what exactly is missing? And can you please post the output of `git status`? (As text, not as screenshot.)

**Comment by ghost** [comment]
Ok thank you. 
I`m trying to copy it from the master branch step-by-step as a new class on the csv-branch...maybe this will work out...we will see

**Comment by ghost** [comment]
@nicolaiparlog 
i deleted the Network.kt Class on my branch...
git_status1
git_status2

**Comment by ghost** [comment]
hmm...that`s not what i wanted...there should be a FIleAccess.kt class and changes in the Main.kt file...

**Comment by nipafx** [comment]
@RalfBr If you stop making changes for a moment, I can help you a bit. Please leave a comment when you have no local changes (and then don't make any).

**Comment by nipafx** [comment]
No local changes means `git status` shows no changes... If you have changes but don't like them, use `git reset --hard HEAD` to get rid of them.

**Comment by ghost** [comment]
```
Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git reset --hard HEAD
HEAD is now at 8c95eb9 commit of Network.kt

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git status
On branch csv
nothing to commit, working tree clean
```

thats what the status is now

**Comment by nipafx** [comment]
Ok, I deleted a file you don't need (`RoadNetwork`) and minimized changes in `Network`. I also saw your changes to `Main` and I think they are good. Returning the `Network` with all the cars is a great idea.

**Comment by nipafx** [comment]
**PLEASE `git pull` BEFORE YOU CONTINUE WORKING.**

**Comment by ghost** [comment]
command "git pull" now in Git? Then next feedback of the actual status?

**Comment by nipafx** [comment]
Now you can get on with `readCarFromCsv` (would be better named `readNetworkFromCsv`). Here's what you need to do (in that order):

1. make it compile by closing all the brackets and returning a network at the end
2. make it more readable by formatting it: right-click on the file in IntelliJ's file tree on the left side and click _Reformat Code_ (somewhere in the middle)
3. make cars (you can find tips on how to do that further above)

**Comment by nipafx** [comment]
Oops, didn't see your message. Yes, `git pull` on your command line, then more coding.

**Comment by ghost** [comment]
Yes did it. Worked!
Thank you for the help!

**Comment by ghost** [comment]
Next Issue with Git :-/
tried to push some changes but it failed...

-------------------------------------------------------------------------------------------------------
Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git status
On branch csv
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   src/main/kotlin/Car.kt
        modified:   src/main/kotlin/Main.kt
        modified:   src/main/kotlin/Network.kt

no changes added to commit (use "git add" and/or "git commit -a")

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git add src/main/kotlin/Main.kt

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git commit -m "reformatted, 'fun readNetworkFromCsv' outside of 'fun main'"
[csv 869f8ad] reformatted, 'fun readNetworkFromCsv' outside of 'fun main'
 1 file changed, 37 insertions(+), 14 deletions(-)

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git push origin "csv"
To https://github.com/RalfBr/project_alpha_omega.git
 ! [rejected]        csv -> csv (non-fast-forward)
error: failed to push some refs to 'https://github.com/RalfBr/project_alpha_omega.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git pull
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> csv


Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git remote -v
origin  https://github.com/RalfBr/project_alpha_omega.git (fetch)
origin  https://github.com/RalfBr/project_alpha_omega.git (push)
upstream        https://github.com/ep6-TH-Bingen/project_alpha_omega (fetch)
upstream        https://github.com/ep6-TH-Bingen/project_alpha_omega (push)

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git push origin "csv"
To https://github.com/RalfBr/project_alpha_omega.git
 ! [rejected]        csv -> csv (non-fast-forward)
error: failed to push some refs to 'https://github.com/RalfBr/project_alpha_omega.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git status
On branch csv
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   src/main/kotlin/Car.kt
        modified:   src/main/kotlin/Network.kt

no changes added to commit (use "git add" and/or "git commit -a")

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git add src/main/kotlin/Car.kt

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git add src/main/kotlin/Network.kt

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git status
On branch csv
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   src/main/kotlin/Car.kt
        modified:   src/main/kotlin/Network.kt


Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git commit -m "Car n Network without changes"
[csv 674b26c] Car n Network without changes
 2 files changed, 5 insertions(+), 6 deletions(-)

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git push origin "csv"
To https://github.com/RalfBr/project_alpha_omega.git
 ! [rejected]        csv -> csv (non-fast-forward)
error: failed to push some refs to 'https://github.com/RalfBr/project_alpha_omega.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git pull
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> csv


Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git status
On branch csv
nothing to commit, working tree clean

Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git pull "csv"
fatal: 'csv' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

---------------------------------------------------------------------------------------------------------

**Comment by nipafx** [comment]
Was ergibt `git log --oneline -5`?

**Comment by ghost** [comment]
Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git log --oneline -5
674b26c Car n Network without changes
869f8ad reformatted, 'fun readNetworkFromCsv' outside of 'fun main'
8c95eb9 commit of Network.kt
276decc copying Network to the branch
d2e8cba added FileAccess.kt , Main.kt changed

**Comment by nipafx** [comment]
My commit c4b3f65 is not in your branch, which means you forgot to `git pull`. Before you can continue to work, you have to fix your local branch:

* save the files you edited in the last two commits by copying them somewhere
* `git branch --set-upstream-to=origin/csv csv` (not requires but makes your life easier)
* `git fetch`
* `git reset --hard origin/csv` (⚠️ this deletes your changes!)

When that is done please once more report back with `git log --oneline -5`.

**Comment by ghost** [comment]
Ralf@Ralf-Lenovo MINGW32 /D/Studium/Info_Abschluss/project_alpha_omega (csv)
$ git log --oneline -5
c4b3f65 Undo unneeded changes
8c95eb9 commit of Network.kt
276decc copying Network to the branch
d2e8cba added FileAccess.kt , Main.kt changed
6e3a252 changed small mistakes

**Comment by nipafx** [comment]
Super, dann kannst du deine gesicherten Änderungen vorsichtig wieder einspielen und weiterarbeiten.

**Comment by ghost** [comment]
Thank you Eden 😄

**Comment by EdenTewelde** [comment]
team work... 💪 

...makes the dream work 😃

File: src/main/kotlin/Main.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import com.univocity.parsers.csv.CsvWriter
+import com.univocity.parsers.csv.CsvWriterSettings
+
+fun main(args: Array<String>) {
+
+    val roadNetwork = readNetworkFromCsv("/cars.csv")
+
+    roadNetwork.analyzeNetwork()
+
+    writeNetworkToCsv(roadNetwork,"Resulting Data")
```
Comment (nipafx) [inline src/main/kotlin/Main.kt]: It's easier if file names contain no spaces. It should also end in `.csv`.

File: src/main/kotlin/Main.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+
```
Comment (nipafx) [inline src/main/kotlin/Main.kt]: Empty lines, so much space... :wink:

File: Resulting Data [hunk 6]
```diff
@@ -0,0 +1,16 @@
+Car-ID,status,delayed
```
Comment (nipafx) [inline Resulting Data]: There's no need to commit this file. In fact, please add this file to `.gitignore` (once you settled on a file name; see below).

**Review by nipafx — CHANGES_REQUESTED** [review]
Good job, almost done.

**Review by nipafx — APPROVED** [review]

**Comment by nipafx** [comment]
Done! :+1:

**Pull request merged**